components/hal/include/hal/uart_hal.h

```diff
@@ -498,6 +488,10 @@ uint16_t uart_hal_get_max_rx_timeout_thrd(uart_hal_context_t *hal);
  */
 #define uart_hal_get_rxfifo_len(hal) uart_ll_get_rxfifo_len((hal)->dev)
 
+#ifdef __cplusplus
```

Comment:
Any reason to move the #fidef up and exclude the last API `uart_hal_get_port_num`?